Actions: skip certain tests on MacOS

MetRonnie opened the pull request:
~Turn off xdist in the hope it makes the integration tests stop hanging flakily on GH Actions~

Reduce homebrew installation and skip style, mypy, in order to speed up MacOS runs

MetRonnie commented:
So as soon as I do this it decides to hang on the "brew install" step instead 🤬

![image](https://user-images.githubusercontent.com/61982285/156027256-098004d0-dbe5-42f1-b0ae-ddf23b3b8095.png)

MetRonnie commented:
Yes, had one today: https://github.com/cylc/cylc-flow/runs/5391678146?check_suite_focus=true

And it's about a 50% failure rate over the last week: https://github.com/cylc/cylc-flow/actions/workflows/test_fast.yml?query=branch%3Amaster

![image](https://user-images.githubusercontent.com/61982285/156399406-cbdf2a44-ed2e-4bfc-abf4-3d7646b8cad3.png)

MetRonnie commented on .github/dependabot.yml:
```diff
@@ -0,0 +1,13 @@
+version: 2
+updates:
+  - package-ecosystem: "pip"
+    directory: "/"
+    schedule:
+      interval: "daily"
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "daily"
+    labels:
+      - 'dependencies'
+      - 'infrastructure'
```
I don't think dependabot can do PEP 517 projects, but at least this will give us updates for GH Actions

oliver-sanders replied:
Cylc-flow isn't PEP517 yet but its deps are stored in the .cfg file, let's see how it gets on...

MetRonnie commented:
Dammit, `brew install` taking an inordinate amount of time suddenly seems to be very common

MetRonnie commented:
Updated to remove the homebrew step completely, think we don't need it

oliver-sanders commented:
> Updated to remove the homebrew step completely, think we don't need it

* shellcheck
  * May be pre-installed, though most likely at an older version.
* sqlite3
  * Don't think this is needed, if something needs it homebrew will install it as a transient dep anyway.
* bash
  * Needed for any Bash interaction, MacOS default shell is zsh so the Bash install is very old.
* coreutils
  * Needed for `ls`, `rm`, that sort of stuff. MacOS has these by default but they are the BSD rather than GNU variants.

So might be able to get away without it but you would have to do some digging to make sure these things aren't needed, the tests passing wouldn't be quite enough.

MetRonnie commented:
- shellcheck: made it skip on MacOS
- sqlite: removed
- bash & coreutils:
  > might be able to get away without it but you would have to do some digging to make sure these things aren't needed, the tests passing wouldn't be quite enough.

  I would argue the tests passing is enough for now, but I guess it might not be in future. I've added them back in, good news is they seem to install without problem

oliver-sanders commented on .github/workflows/test_fast.yml:
```diff
@@ -77,8 +83,10 @@ jobs:
+        timeout-minutes: 6
         run: |
-          pytest tests/integration
+          pytest -n 0 --dist=no tests/integration
+        # Turned off x-dist due to flakiness on GH Actions for some reason
```
Would rather not change something just to cover up issues at our end, at least until we understand what the issues are?

Can we at least try with just the other fixes first and see whether this is necessary after?

```suggestion
          pytest tests/integration
```

MetRonnie replied:
Removed this change

oliver-sanders commented on .github/workflows/test_fast.yml:
```diff
@@ -77,8 +83,10 @@ jobs:
           bandit -r --ini .bandit cylc/flow
 
       - name: Integration Tests
+        timeout-minutes: 6
```
This might be a little on the tight side?

oliver-sanders replied:
Fair enough.

oliver-sanders commented:
Most recent run timed out: https://github.com/cylc/cylc-flow/runs/5451461210?check_suite_focus=true#step:13:71

oliver-sanders reviewed (approved):
Ok, let's see what dependabot does for Python.

MetRonnie commented:
Checks haven't yet run since I pushed, probably due to https://www.githubstatus.com/incidents/tyc8wpsgr2r8

MetRonnie commented:
Looks like it's done the trick, in the most recent run, ubuntu/py3.7 hung instead of macos

MetRonnie commented:
I've also added uploading of the cylc-run directory as an artifact if `test_fast.yml` fails, hopefully that might help us debug next time it hangs
